app/portainer/home/EnvironmentList/KubeconfigButton.tsx

```diff
@@ -34,48 +81,154 @@ export function KubeconfigButton({ environments }: Props) {
       category: 'kubernetes',
     });
 
-    showKubeconfigModal(environments);
+    selection.clear();
+    setShowDialog(true);
+  }
+
+  function handleClose() {
+    setShowDialog(false);
+  }
+
+  function handleDownload() {
+    confirmKubeconfigSelection();
+  }
+
+  function isKubeconfigButtonVisible(environments: Environment[]) {
+    if (window.location.protocol !== 'https:') {
+      return false;
+    }
+    return environments.some((env) => isKubernetesEnvironment(env.Type));
+  }
+
+  function useKubeServiceExpiry() {
```

Comment:
well, now it's more obvious, but the expiry message is for kubeconfig, so should be `useKubeconfigExiryMessage`

Reply:
Use query directly.